packages/widgets/src/blocks/legacy-widget/edit/preview.js

```diff
@@ -69,6 +70,9 @@ export default function Preview( { idBase, instance, isVisible } ) {
 					iframe.removeEventListener( 'load', setHeight );
 				};
 			}
+			renderPreview( idBase, instance ).then( ( response ) => {
+				iframe.srcdoc = response.preview;
```

Comment:
It's probably better to set state here instead of manipulating DOM directly. We'll have to create a new state and call the setState function here.

```js
const [ srcDoc, setSrcDoc ] = useState( '' );

renderPreview( idBase, instance ).then( ( response ) => {
  setSrcDoc( response.preview );
} );
```

We'll probably also want to add some error handling and abort the fetch when the component is unmounted too. Which can be a follow-up PR.